src/components/Input/Input.tsx

```diff
@@ -106,12 +105,7 @@ export const Input = React.forwardRef<HTMLInputElement, Props>(
                         readOnly,
                     })}
                 >
-                    {leadingIcon &&
-                        React.cloneElement(leadingIcon.icon, {
-                            viewBox: '0 0 24 24',
-                            ...elem('icon', { type: 'leading', bold: hasText, size }),
-                            onClick: leadingIcon.callback ? leadingIcon.callback : undefined,
-                        })}
+                    {leadingIcon || <></>}
```

Comment:
Why do we need this fallback? I mean if `leadingIcon: ReactElement` then it should render in the DOM ok. What am I missing?